src/compiler/checker.ts

```diff
@@ -21564,14 +21596,107 @@ namespace ts {
                 return !assumeTrue;
             }
 
-            function narrowByInKeyword(type: Type, literal: LiteralExpression, assumeTrue: boolean) {
-                if (type.flags & (TypeFlags.Union | TypeFlags.Object)
+            function widdenTypeWithSymbol(type: Type, newSymbol: Symbol): Type {
+                // If type is this/any/unknown, it could not be widden.
+                if ((type.flags & TypeFlags.AnyOrUnknown) || isThisTypeParameter(type)) {
+                    return type;
+                }
+                const propName = newSymbol.escapedName;
+                const members = createSymbolTable();
+                members.set(propName, newSymbol);
+                const newObjType = createWidenType(/* symbol */ undefined, members, emptyArray, emptyArray, /* stringIndexInfo */ undefined, /* numberIndexInfo */ undefined);
+
+                // if `type` is never, just return the new anonymous object type.
+                if (type.flags & TypeFlags.Never) {
+                    return newObjType;
+                }
+
+                // if type is intersection, we might have added type into it, and we just need to add into this type again rather than a new one.
+                // else add a new anonymous object type which contains the type and widden the origional type with it.
+
+                if (isIntersectionType(type)) {
+                    // try to get the first Anonymous Object type to add new type to it.
+                    const widenedType: Type | undefined = type.types.find(t => isObjectType(t) && t.objectFlags & ObjectFlags.WidenedByNarrow);
+                    if (widenedType && isObjectType(widenedType)) {
+                        const typeWithOutWiden = filterIntersectionType(type, t => t !== widenedType);
+
+                        const members = createSymbolTable();
+                        members.set(propName, newSymbol);
+                        if (widenedType.members) {
+                            mergeSymbolTable(members, widenedType.members);

```

Comment:
This recursively merges the symbol table deeply (so overlapping symbols within the table get combined, and have a merge pointer at the new, combined symbol). You don't want that. Rather than most of this logic, you should _probably_ just use `getSpreadType`, rather than everything after the `// if type is intersection` comment, since it already combines object types where appropriate, since you guarantee you're adding a new property in the second object anyway. (Re)Using it should also allow you to cut out a lot of the other structural changes around intersection construction in this PR, too.